sdgx/data_models/metadata.py

```diff
@@ -13,22 +13,48 @@
 from sdgx.exceptions import MetadataInitError
 from sdgx.utils import logger
 
-# TODO: Design metadata for relationships...
-# class DType(Enum):
-#     datetime = "datetime"
-#     timestamp = "timestamp"
-#     numeric = "numeric"
-#     category = "category"
 
+class Metadata(BaseModel):
+    """Metadata
 
-# class Relationship:
-#     pass
+    This metadata is mainly used to describe the data types of all columns in a single data table.
 
+    For each column, there should be an instance of the Data Type object.
 
-class Metadata(BaseModel):
+    Args:
+        primary_key(str): The primary key, a field used to uniquely identify each row in the table.
+        The primary key of each row must be unique and not empty.
+
+        composite_primary_key(bool): Whether to enable the composite primary key feature.
+
+        primary_key_list(bool): List of composite primary keys.
+
+        column_list(list[str]): list of the comlumn name in the table, other columns lists are used to store column information.
+    """
+
+    # for primary key
+    # compatible with single primary key or composite primary key
+    primary_key: str = "primary_key undefined"
+    _composite_primary_key: bool = False
+    primary_key_list: list = []
```

Comment:
Why don't we just use a list(`primary_keys`) for single or composite primary key?

```suggestion
    primary_keys: List[str] = []
```

and use `len(primary_keys) == 1` for single's situation.


Another way is 

```suggestion
    primary_key: Union[str, List[str]]= ""
```

and use `isinstance` for telling single or composite